### ISSUE TITLE
[Android] 'showModal': Screen is re-opened after returning from home

### ISSUE THREAD

**Issue by kyryloz** [wix/react-native-navigation#2639]
_First of all I want to thank you all, guys, for such wonderful navigation library. After frustrating experience with `react-navigation` I feel like I'm able to breathe freely again. This is how navigation should be implemented on RN, I'm disappointed that Facebook points us to the `react-navigation` as the primary navigation library. There should be you, guys, please, keep going!_

### Issue Description
Screen is opened with 'showModal'. After pressing Home button and then return to the app the open screen animation is re-played. It's not actually re-opened again, but gives the impression that it is. I believe it's because 'modal' screens on Android is implemented as dialogs (please, correct me if I'm wrong), so the system hides them after user presses Home button and restore – when returns to the app.

### Steps to Reproduce / Code Snippets / Screenshots
The code:
```javascript
import React from 'react'
import {
  Button,
  Text,
  View,
} from 'react-native'
import { Navigation } from 'react-native-navigation'

const FirstScreen = ({ navigator }) => (
  <View>
    <Text>First screen</Text>
    <Button
      onPress={() => {
        navigator.showModal({
          screen: 'example.SecondScreen',
          title: 'Second screen',
        })
      }}
      title="Open second screen"
    />
  </View>
)

const SecondScreen = () => (
  <View style={{ flex: 1, backgroundColor: 'green' }}>
    <Text>Second screen</Text>
  </View>
)

Navigation.registerComponent('example.FirstScreen', () => FirstScreen)
Navigation.registerComponent('example.SecondScreen', () => SecondScreen)

Navigation.startTabBasedApp({
  tabs: [
    {
      label: 'First',
      screen: 'example.FirstScreen',
      title: 'Screen One',
      icon: require('./ic_launcher.png'),
    },
  ],
})
```
That's it.

Steps:
1. Open screen with 'showModal'.
2. Press home button.
3. Return to the app.

The gif should be pretty much explanatory. I've reduced animation speed on the emulator so we can see what's going on in slow motion:

![issue](https://media.giphy.com/media/l3diLqiWe36QcXkyc/giphy.gif)

---
### Environment
* React Native Navigation version: wix/react-native-navigation#2580/head (since stable version is not working with RN 0.52, but I believe it's reproduces on 1.1.366 as well)
* React Native version: 0.52.2
* Platform: Android
* Device info: Android Emulator (API 27); Nexus 7 (API 23) – Debug

**Comment by kyryloz** [comment]
Seems like `animationType: 'screen'` is working good for modals. But this type of animation does not mentioned in the docs. 

Regarding my note about opening new screen as fullscreen without using modals. I've found `StyleParams#drawScreenAboveBottomTabs` param, and related code:

```java
private LayoutParams createScreenLayoutParams(ScreenParams params) {
        LayoutParams lp = new LayoutParams(MATCH_PARENT, MATCH_PARENT);
        if (params.styleParams.drawScreenAboveBottomTabs) {
            lp.addRule(RelativeLayout.ABOVE, bottomTabs.getId());
        }
        return lp;
    }
```
Which, as I understand, if set to `false` should draws new pushed View as fullscreen View. But in reality this code does nothing, all pushed View's are always drawn above the tabs. Is it intended behavior?

**Comment by pqkluan** [comment]
Sorry If I get your idea wrong but isn't this achievable with push and this navigatorStyle prop?
```jsx
tabBarHidden: false, // make the screen content hide the tab bar (remembered across pushes)
```

**Comment by kyryloz** [comment]
Yeah, you’re right, it works perfectly on iOS (when new screen is pushed the tabs stays on the previous screen), but on Android the tabs hides with an animation, and it looks a little weird. Even if I hide tabs w/o animation it doesn’t look good, from the user experience it looks like we open new screen and tabs on previous screen suddenly disappear and appear again after the user presses back. 

I can make a gif to make things more clearly.

Actually, as far as I’m concerned, to achieve more natural animation we don’t need to hide tabs. Instead, Screen (Android View) should takes all the space in the root RelativeLayout. But tabs are always on top and I can’t figure out why. :)

**Comment by pqkluan** [comment]
You could also set the animation to `fade` for better visual.

If that can't also satisfy you, there are two other choices:
- Fork the repo and custom the native animation to your liking
- Wait for a v2 feature that supports custom animation (soon™)

**Comment by kyryloz** [comment]
Yeah, I understand. Thank you for your support!
